query.go

```diff
@@ -1202,15 +1214,31 @@ func (s *ChainService) sendTransaction(tx *wire.MsgTx, options ...QueryOption) e
 		// Then return the first error we have for that code (it doesn't
 		// really matter which one, as long as our error code parsing is
 		// correct).
-		for _, broadcastErr := range rejections {
-			if broadcastErr.Code == mostRejectedCode {
-				return broadcastErr
-			}
-		}
+		return firstRejectWithCode(mostRejectedCode)
+	}
 
-		// We can't really get here unless something is totally wrong in
-		// the above error mapping code.
-		return fmt.Errorf("invalid error mapping")
+	// We did get some rejections, but not from all peers. Perhaps that's
```

Comment:
This is intended to _not_ return an error if say only one of our peers rejects (since we wait longer now?) the transaction?

Reply:
This is intended to return an error in slightly more cases. Before, we'd only return an error if all the peers responded with an error. Now we also return an error if only 50% of the peers reject.

That doesn't handle the case where only one peer rejects (the case where we're connected to a bunch of `bitcoind` nodes and just one `btcd`). Not sure what to do there, since we also don't want to open ourselves up to a malicious peer always rejecting and us failing because of that.
Perhaps we need to take an even closer look at the _type_ of error we get? Like if we get one "input already spent", we give that 100% weight as with other rejects we use a threshold?

Reply:
> Not sure what to do there, since we also don't want to open ourselves up to a malicious peer always rejecting and us failing because of that.

Yeah there's not really a great solution here...which I think was part of the rationale that the bitcoind devs used 

Looking at the type of error would certainly help here. Ultimately, I think our rebroadcaster would help to fill the gap here: as long as we keep rebroadcasting until the transaction has been _confirmed_, then things should _eventually_ be mined.